src/core/qgspointlocator.cpp

```diff
@@ -574,27 +586,107 @@ void QgsPointLocator::setRenderContext( const QgsRenderContext *context )
 
 }
 
-bool QgsPointLocator::init( int maxFeaturesToIndex )
+void QgsPointLocator::onInitTaskTerminated()
 {
-  return hasIndex() ? true : rebuildIndex( maxFeaturesToIndex );
+  mIsIndexing = false;
+  mRenderer.reset();
+  mSource.reset();
 }
 
+void QgsPointLocator::onRebuildIndexFinished( bool ok )
+{
+  onInitTaskTerminated();
+
+  // treat added and deleted feature while indexing
+  for ( QgsFeatureId fid : mAddedFeatures )
+    onFeatureAdded( fid );
+
+  for ( QgsFeatureId fid : mDeletedFeatures )
+    onFeatureDeleted( fid );
```

Comment:
Do these arrays of added/deleted features ever get cleared?

Reply:
Done